docker/Dockerfile

```diff
@@ -53,13 +54,23 @@ COPY *.mk .
 # Build
 ARG VERSION="unknown"
 ARG GIT_COMMIT="unknown"
-RUN make gpu-operator
+RUN make cmd-gpu-operator
+RUN make cmd-validator
+
+FROM ${CUDA_SAMPLE_IMAGE} AS sample-builder
+RUN mkdir /artifacts
+RUN cp /cuda-samples/vectorAdd /artifacts/vectorAdd
 
 FROM nvcr.io/nvidia/cuda:12.9.0-base-ubi9
 
 # Remove CUDA libs(compat etc) in favor of libs installed by the NVIDIA driver
 RUN dnf remove -y cuda-*
 
+RUN dnf install -y \
+        kmod \
+        pciutils && \
+    rm -rf /var/cache/yum/*
```

Comment:
We should be able to remove this once we move to the distroless base image in https://github.com/NVIDIA/gpu-operator/pull/1417.

The distroless `dev` tags comes with busybox, which include the utilities needed from these packages, namely `lsmod` and `lspci`. 

For some historical context, `kmod` was added to the validator image in https://github.com/NVIDIA/gpu-operator/commit/24c7bd0d7b32d81dd6507877f49e45b4c04b5fdb specifically to run `lsmod`. `pciutils` was added in https://github.com/NVIDIA/gpu-operator/commit/c657465f44aa332da16da5ed05bd7f55722abdf8 specifically to run `lspci` [here](https://github.com/NVIDIA/gpu-operator/blob/dfbb6c5c4464723f6f1e4c736d3e9c3e2660deed/validator/metrics.go#L248).